src/DomainGroupResolver.php

```diff
@@ -54,7 +40,7 @@ class DomainGroupResolver implements DomainGroupResolverInterface {
    * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
    *   Current entity repository interface.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, DomainNegotiatorInterface $domain_negotiator, RouteMatchInterface $current_route_match, EntityRepositoryInterface $entity_repository) {
+  public function __construct(protected EntityTypeManagerInterface $entity_type_manager, protected DomainNegotiatorInterface $domain_negotiator, protected RouteMatchInterface $current_route_match, protected EntityRepositoryInterface $entity_repository) {
```

Comment:
If we're declaring theme here (`protected EntityTypeManager Interface ...` etc) we should switch to `protected EntityTypeManagerInterface $entityTypeManager` however if the Trait is declaring them I suspect we need to just remove our declaration of them, and set them, not declaring `$entity_type_manager` as a $this variable.

Reply:
> not declaring $entity_type_manager as a $this variable.
Right.

Reply:
> I suspect we need to just remove our declaration of them

We have two options here.  We can keep the declarations or remove them.  I am not sure of the merit of either.  You seem to prefer to remove the declaration.s  So I will just go with that.